### PR TITLE
minor-improvements

### PR DESCRIPTION
**Pull request by AinGrace**
- Removed unnecessary null and Exception checks
- Replaced return value of JwtUtility.extractJWT() and UserService.parseJWT() from Result to Optional
- Removed logs from repos and moved them to application layer to reduce verbosity
- Fixed Regex
- Reduced the complexity of Argon2.hash() for the sake of performance
- Banner

### CODE REVIEW

File: src/main/java/core/project/chess/domain/aggregates/user/value_objects/Username.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 public record Username(String username) {
 
-    private static final String usernameRegex = "^[a-zA-Z0-9 ]*$";
+    private static final String usernameRegex = "^[a-zA-Z0-9]*$";
```
Comment (AinGrace) [inline src/main/java/core/project/chess/domain/aggregates/user/value_objects/Username.java:9]: br space in regex makes no sense

File: src/main/java/core/project/chess/domain/aggregates/user/entities/EmailConfirmationToken.java [hunk 2]
```diff
@@ -17,11 +17,6 @@ public class EmailConfirmationToken {
     private final /**@OneToOne*/ UserAccount userAccount;
 
     private EmailConfirmationToken(UUID tokenId, Token token, TokenEvents tokenEvents, boolean isConfirmed, UserAccount userAccount) {
-        Objects.requireNonNull(tokenId);
-        Objects.requireNonNull(token);
-        Objects.requireNonNull(tokenEvents);
-        Objects.requireNonNull(userAccount);
-
         this.tokenId = tokenId;
```
Comment (AinGrace) [inline src/main/java/core/project/chess/domain/aggregates/user/entities/EmailConfirmationToken.java:20]:
```suggestion
Objects.requireNonNull(tokenId);
```

File: src/main/java/core/project/chess/domain/aggregates/user/entities/EmailConfirmationToken.java [hunk 3]
```diff
@@ -17,11 +17,6 @@ public class EmailConfirmationToken {
     private final /**@OneToOne*/ UserAccount userAccount;
 
     private EmailConfirmationToken(UUID tokenId, Token token, TokenEvents tokenEvents, boolean isConfirmed, UserAccount userAccount) {
```
Comment (AinGrace) [inline src/main/java/core/project/chess/domain/aggregates/user/entities/EmailConfirmationToken.java:19]:
```suggestion
Objects.requireNonNull(tokenId);
Objects.requireNonNull(token);
Objects.requireNonNull(tokenEvents);
Objects.requireNonNull(userAccount);

**Review by HaciHaciyev — APPROVED** [review]
I appreciate your refactoring

**Pull request closed without merging**